Auth0/Auth0Authentication.swift

```diff
@@ -245,4 +253,26 @@ struct Auth0Authentication: Authentication {
             .connection(connection)
     }
     #endif
+    
+    // MARK: - Private Methods
+    
+    // swiftlint:disable:next function_parameter_count
+    private func login(username: String, otp: String, realm: String, audience: String?, scope: String?, parameters: [String: Any]) -> Request<Credentials, AuthenticationError> {
+        let url = URL(string: "/oauth/token", relativeTo: self.url)!
+        var payload: [String: Any] = [
+            "username": username,
+            "otp": otp,
+            "realm": realm,
+            "grant_type": "http://auth0.com/oauth/grant-type/passwordless/otp",
+            "client_id": self.clientId
+        ]
+        if let audience = audience {
+            payload["audience"] = audience
+        }
+        if let scope = scope {
+            payload["scope"] = scope
```

Comment:
Should we have a default scope if none is provided? Is there a default scope when logging in via browser?

Our SDKs usually fall back to `"openid email profile"`. If this is similar to an RO grant (trading credentials for tokens), then I would consider a default here.

Reply:
There is a default scope (`openid`), it's in the default implementation of each login method of the `Authentication` protocol. Since in Swift protocol method parameters can't have default values, you have to add a default implementation _in_ the protocol providing default values:

https://github.com/auth0/Auth0.swift/pull/330/files#diff-629bd26385f909338bd67e4e7ff962b1R676

I've used `openid` as the default because that's the default value in the Android implementation:

https://github.com/auth0/Auth0.Android/pull/271/files#diff-2c8621d496cf40311246e1300b4b852fL322